mlrun/api/api/utils.py

```diff
@@ -470,6 +471,76 @@ def process_function_service_account(function):
     )
 
 
+def ensure_function_security_context(function, auth_info: mlrun.api.schemas.AuthInfo):
+    """
+    For iguazio we enforce that pods run with user id and group id depending on
+    mlrun.mlconf.function.spec.security_context.enrichment_mode
+    and mlrun.mlconf.function.spec.security_context.enrichment_group_id
+    """
+    function: mlrun.runtimes.pod.KubeResource
+
+    # if security context is not required.
+    # security context is not yet supported with spark runtime since it requires spark 3.2+
+    if (
+        mlrun.mlconf.function.spec.security_context.enrichment_mode
+        == SecurityContextEnrichmentModes.disabled.value
+        or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
+        or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        or mlrun.mlconf.httpdb.authentication.mode != "iguazio"
```

Comment:
I believe our common way to recognize we're running inside Iguazio system (which probably should be moved to a method) is to check whether `mlrun.mlconf.igz_version` is set and not by the authentication mode